Makefile

```diff
@@ -0,0 +1,118 @@
+# I don't know Golang but I've heard of people who do
+# Vaguely copied from https://github.com/jessfraz/gitable/blob/master/Makefile
+
+# Set an output prefix, which is the local directory if not specified
+PREFIX?=$(shell pwd)
+
+# Setup name variables for the package/tool
+NAME := vltgatekeeper
+PKG := github.com/praekeltfoundation/vault-gatekeeper-mesos
+
+# Set any default go build tags
+BUILDTAGS :=
+
+# Set the build dir, where built cross-compiled binaries will be output
+BUILDDIR := ${PREFIX}/build
+
+# Populate version variables
+# Add to compile time flags
+VERSION := $(shell cat VERSION.txt)
+GITCOMMIT := $(shell git rev-parse --short HEAD)
+GITUNTRACKEDCHANGES := $(shell git status --porcelain --untracked-files=no)
+ifneq ($(GITUNTRACKEDCHANGES),)
+	GITCOMMIT := $(GITCOMMIT)-dirty
+endif
+CTIMEVAR=-X $(PKG)/version.GITCOMMIT=$(GITCOMMIT) -X $(PKG)/version.VERSION=$(VERSION)
+GO_LDFLAGS=-ldflags "-w $(CTIMEVAR)"
+GO_LDFLAGS_STATIC=-ldflags "-w $(CTIMEVAR) -extldflags -static"
+
+# List the GOOS and GOARCH to build
+# TODO: Decide on more architectures if necessary
+GOOSARCHES = darwin/amd64 linux/amd64
+
+# This bootstrap stuff is copied from Vault's makefile:
+# 
+EXTERNAL_TOOLS = \
+	github.com/golang/dep/cmd/dep \
+	golang.org/x/lint/golint
+
+.PHONY: bootstrap
+bootstrap:
+	@for tool in  $(EXTERNAL_TOOLS) ; do \
+		echo "Installing/Updating $$tool" ; \
+		go get -u $$tool; \
+	done
+
+.PHONY: build
+build: $(NAME) ## Builds a dynamic executable or package
+
+$(NAME): *.go VERSION.txt
+	@echo "+ $@"
+	go build -tags "$(BUILDTAGS)" ${GO_LDFLAGS} -o $(NAME) .
+
+.PHONY: static
+static: ## Builds a static executable
+	@echo "+ $@"
+	CGO_ENABLED=0 go build \
+				-tags "$(BUILDTAGS) static_build" \
+				${GO_LDFLAGS_STATIC} -o $(NAME) .
+
+all: clean build fmt lint test vet install ## Runs a clean, build, fmt, lint, test, vet and install
+
+.PHONY: fmt
+fmt: ## Verifies all files have men `gofmt`ed
+	@echo "+ $@"
+	@gofmt -s -l . | grep -v vendor
+
+.PHONY: lint
+lint: ## Verifies `golint` passes
+	@echo "+ $@"
+	@golint ./... | grep -v vendor
+
+.PHONY: test
+test: ## Runs the go tests
+	@echo "+ $@"
+	@go test -v -tags "$(BUILDTAGS) cgo" $(shell go list ./... | grep -v vendor)
+
+.PHONY: vet
+vet: ## Verifies `go vet` passes
+	@echo "+ $@"
+	@go vet $(shell go list ./... | grep -v vendor)
+
+.PHONY: cover
+cover: ## Runs go test with coverage
+	@echo "" > coverage.txt
+	@for d in $(shell go list ./... | grep -v vendor); do \
+		go test -race -coverprofile=profile.out -covermode=atomic "$$d"; \
```

Comment:
This doesn't stop if it fails :(

Reply:
Maybe we need to collect exit codes and return the highest or something? I don't know what support `make` has for that, though.